code/datums/components/quality.dm

```diff
@@ -0,0 +1,78 @@
+#define UNUSABLE_QUALITY 0.7
+#define SHODDY_QUALITY 0.8
+#define POOR_QUALITY 0.9
+#define NORMAL_QUALITY 1
+#define GOOD_QUALITY 1.05
+#define FINE_QUALITY 1.1
+#define SUPERIOR_QUALITY 1.2
+#define EXCEPTIONAL_QUALITY 1.3
+#define MASTERFUL_QUALITY 1.5
+#define ARTIFACT_QUALITY 1.75 // big upgrade coz it is so rare
+
+datum/component/quality
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+
+	var/list/quality_levels = list(UNUSABLE_QUALITY,SHODDY_QUALITY,POOR_QUALITY,NORMAL_QUALITY,GOOD_QUALITY,FINE_QUALITY,SUPERIOR_QUALITY,EXCEPTIONAL_QUALITY,MASTERFUL_QUALITY,ARTIFACT_QUALITY )
+	var/quality_level
+	var/old_name
+
+	var/obj/item/parent_item
+
+	var/mob/living/carbon/human/creator
+	var/datum/mind/creator_mind
+	var/skill_level
```

Comment:
Why would this component hold onto these references?

Reply:
It should really just hold the creator so it can say it was made by a certain player, the mind and level can be found on the player.